lms/djangoapps/commerce/signals.py

```diff
@@ -0,0 +1,146 @@
+        request_user: the user as whom to authenticate to the commerce service
+            when attempting to initiate the refund.
+
+    Returns:
+        A list of the external service's IDs for any refunds that were initiated
+            (may be empty).
+
+    Raises:
+        exceptions.SlumberBaseException: for any unhandled HTTP error during
+            communication with the commerce service.
+        exceptions.Timeout: if the attempt to reach the commerce service timed
+            out.
+
+    """
+    course_key_str = unicode(course_enrollment.course_id)
+    unenrolled_user = course_enrollment.user
+
+    try:
+        refund_ids = ecommerce_api_client(request_user or unenrolled_user).refunds.post(
+            course_id=course_key_str,
+            username=unenrolled_user.username,
+        )
+    except HttpClientError, exc:
+        if exc.response.status_code == 403 and request_user != unenrolled_user:
+            # this is a known limitation; commerce service does not presently
+            # support the case of a non-superusers initiating a refund on
+            # behalf of another user.
+            log.warning("User [%s] was not authorized to initiate a refund for user [%s] "
+                        "upon unenrollment from course [%s]", request_user, unenrolled_user, course_key_str)
+            return []
+        else:
+            # no other error is anticipated, so re-raise the Exception
+            raise exc
+
+    if refund_ids:
+        # at least one refundable order was found.
+        log.info(
+            "Refund successfully opened for user [%s], course [%s]: %r",
+            unenrolled_user.id,
+            course_key_str,
+            refund_ids,
+        )
+        try:
+            send_refund_notification(course_enrollment, refund_ids)
+        except:  # pylint: disable=bare-except
+            # don't break, just log a warning
+            log.warning("Could not send email notification for refund.", exc_info=True)
+    else:
+        # no refundable orders were found.
+        log.debug("No refund opened for user [%s], course [%s]", unenrolled_user.id, course_key_str)
+
+    return refund_ids
+
+
+def send_refund_notification(course_enrollment, refund_ids):
+    """
+    Issue an email notification to the configured email recipient about a
+    newly-initiated refund request.
+
+    This function does not do any exception handling; callers are responsible
+    for capturing and recovering from any errors.
+    """
+    if microsite.is_request_in_microsite():
+        # this is not presently supported with the external service.
+        raise NotImplementedError("Unable to send refund processing emails to microsite teams.")
+
+    for_user = course_enrollment.user
+    subject = _("[Refund] User-Requested Refund")
+    message = _("A refund request has been initiated for {0} ({1}). "
+                "To process this request, please visit the link(s) below.").format(for_user, for_user.email)
```

Comment:
i think str(for_user) gives back something human readable (if only because this code is copy pasted from shoppingcart).  Regardless, I'm making the change you suggested.